cookbooks/bcpc/recipes/graphite.rb

```diff
@@ -21,7 +21,23 @@
 include_recipe "bcpc::apache2"
 
 make_config('mysql-graphite-user', "graphite")
-make_config('mysql-graphite-password', secure_password)
+# backward compatibility
+mysql_graphite_password = get_config("mysql-graphite-password")
+if mysql_graphite_password.nil?
+  mysql_graphite_password = secure_password
+end
+
+bootstrap = get_all_nodes.select{|s| s.hostname.include? 'bootstrap'}[0].fqdn
+
+nodes = get_nodes_for("graphite").map!{ |x| x['fqdn'] }.join(",")
```

Comment:
Instead of doing a `map!` on `get_nodes_for`, please use the node attribute function `[get_node_attributes](https://github.com/bloomberg/chef-bach/blob/94b5c20e7ed967e111daae6c05d15cb5642e381a/cookbooks/bcpc/libraries/utils.rb#L176-L180)`